### DISCUSSION TITLE
Add Disabled State to Options

### DISCUSSION THREAD

**Discussion by davidmdm** [charmbracelet/huh#584] [Ideas]
**Is your feature request related to a problem? Please describe.**
We want to provide multiselects to users where some of the options are disabled.
Generally speaking we can filter our disabled options but for our use-case it would feel like missing context for our users.
It would be better if they could see the state of all their options even if specific options have been disabled from selection.

**Describe the solution you'd like**
It would be great if we could do:
```go
 huh.NewMultiSelect[string]().
    Title("Toppings").
    Options(
        huh.NewOption("Lettuce", "lettuce").Selected(true),
        huh.NewOption("Tomatoes", "tomatoes").Disabled(true),
        ...
```

The result would be that Tomatoes may appear gray and/or strikethrough and could not be selected.
Perhaps the arrow functions would skip over its index entirely, or selection would be blocked or trigger some visual element.
I have no hard requirements as to user experience.

**Comment by maaslalani** [comment]
Hey! Thanks so much for creating this issue!

I'm curious on what your exact use case is here. Why would you prefer an option to be disabled rather simply removing it from the list. We don't mind adding something like this to `huh`, but it would be awesome to have a very compelling example that we can show off in the documentation.

**Comment by davidmdm** [comment]
Essentially when you remove an item from a list, users sometimes think your application is broken and that items they expect to be present are missing. It's better to be able to display the items, but make them unselectable / styled differently, and so forth.

**Comment by gabrielchl** [comment]
An example to what david was pointing out would be:
```
Toppings:
> [ ] Lettuce
  [ ] Tomatoes
  [ ] Charm Sauce (sorry, we've run out!)
  [ ] Nutella
```
where we wan't to let the users know that the "Charm Sauce" option exists, but is just temporarily disabled in this case

**Comment by bashbunni** [comment]
Hey @davidmdm , thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.

Thank you @gabrielchl for the great example on this.
